### ISSUE TITLE
diego cell: inconsistent app instances running when changing instance count

### ISSUE THREAD

**Issue by munzli** [SUSE/scf#2372]
***Problem***
Downgrading instance count through a PUT request on `/v2/apps/<uuid>` while the app is running leaves old and outdated app processes running on the diego cells.

This happens while pushing and scaling apps and even leaves apps running after deleting/stopping them. ***If pushing a different version you will receive different apps on the same route!***

Restarting the diego cells/brain or killing the processes on the diego cell pods manually with `kill` helps.

***Installed Versions***
SCF installed through helm with version 2.15.2
Sizing doesn't seem to matter

***How to reproduce***
Create a simple static page zipped app. *Use the script below.*
Push the app with more than 1 instance
`cf push test-app -p hello.zip -i 5`

To better see the problem change the text in the index.html and repush the app without defining instances:
`cf push test-app -p hello.zip`

It still says that the app has 5 instances, but since the manifest defines 3 there will be an inconsistency in the running apps.

If you try to delete or stop the app it will also still be available (the first version) when calling the route.

Short script to create a little static app:
```
#!/usr/bin/env sh

touch /tmp/Staticfile

cat > /tmp/manifest.yml <<EOF
---
applications:
    - name: static
      buildpack: staticfile_buildpack
      instances: 3
      memory: 64M
      disk_quota: 64MB
EOF

cat > /tmp/index.html <<EOF
<!DOCTYPE html>
<html>

<head>
    <meta charset="UTF-8">

<body>
	Simple Nr1
</body>

</html>
EOF

zip -j hello.zip /tmp/Staticfile /tmp/manifest.yml /tmp/index.html

rm /tmp/Staticfile /tmp/manifest.yml /tmp/index.html
```

*Note* 
The problem doesn't exist when changing the `scale` or `push` command in the cf-cli code to stop the app before calling `cmd.appRepo.Update`.

**Comment by mook-as** [comment]
Hi there!

I believe this is because we broke cc-clock during the switch to BPM, and should be fixed by #2193 (which landed in 2.16.0).  Please let us know if that sounds correct, thanks!

**Comment by adrianliechti** [comment]
ciao @mook-as,

do you provide helm-charts for pre-releases? this would drastically ease us to test the fix.

best, adrian

**Comment by troytop** [comment]
@adrianliechti right now we don't have publicly available pre-releases, though we'll consider ways to deliver these in the future.

The good news is that 2.16.4 has been released to the SUSE registry and chart repo.

**Comment by adrianliechti** [comment]
haha good one!
we will update our test clusters after a short sleep and give feedback asap!
(we are also super curious for the eirini preview)

**Comment by adrianliechti** [comment]
okay... short update (couldn't sleep...) I've updated a first cluster (AKS, 6 Nodes)

- helm upgrade seems to work flawless
- First good news: the apps are started again after upgrade
  (on the upgrade to 2.15.2, we needed to restage the apps: cf state was telling started - but actually they were not and never did without manually restage them.). this seems better now

hope sleeping now works as well as expected :)

**Comment by adrianliechti** [comment]
the feedback about the stability from our developers is very good after the upgrade!

PS: your eirini support is super cool :)

**Comment by andreas-kupries** [comment]
Ok. As things seem to have been fixed with the switch 2.16.4 and no changes in nearly a week I will now close this issue.
